### PR TITLE
Fix scalar as_lazy_data

### PR DESCRIPTION
**Pull request by djkirkham**
This is a workaround for https://github.com/dask/dask/issues/2823 (now fixed on dask/master), identified in https://github.com/SciTools/iris/pull/2861.

### CODE REVIEW

**Comment by cpelley** [comment]
Is it worth adding a test that captures current dask behaviour?  That way, when it fails (iris uses a dask release with the fix), it will be obvious to remove the workaround.

**Comment by djkirkham** [comment]
@cpelley That makes sense

File: lib/iris/tests/unit/lazy_data/test_as_concrete_data.py [hunk 1]
```diff
@@ -62,6 +75,35 @@ def test_lazy_mask_data(self):
...
+        a = np.ma.masked_array(5, True)
+        proxy = MyProxy(a)
+        lazy_array = as_lazy_data(proxy)
+        self.assertTrue(is_lazy_data(lazy_array))
+        result = as_concrete_data(lazy_array)
+        self.assertFalse(is_lazy_data(result))
+        self.assertMaskedArrayEqual(result, a)
+
+    def test_dask_scalar_proxy_pass_through(self):
+        # This test will fail when using a version of Dask with
```
Comment (pelson) [inline lib/iris/tests/unit/lazy_data/test_as_concrete_data.py]: A little bit naughty (hello future person who is going back and reading this PR).

File: lib/iris/_lazy_data.py [hunk 2]
```diff
@@ -89,6 +89,15 @@ def _limited_shape(shape):
     return tuple(shape)
 
 
+def _getall(a):
+    res = a[()]
+    if isinstance(res, ma.core.MaskedConstant):
+        res = ma.masked_array(res.data, mask=res.mask)
+    return res
+
+_getall_delayed = dask.delayed(_getall)
```
Comment (pelson) [inline lib/iris/_lazy_data.py]: There is a decorator for this purpose...

File: lib/iris/tests/unit/lazy_data/test_as_concrete_data.py (hunk 1 above)
Comment (djkirkham) [reply lib/iris/tests/unit/lazy_data/test_as_concrete_data.py]: Hello

**Pull request merged**